### PR TITLE
Feature/filter home screen workouts

### PR DESCRIPTION
**Pull request by Alex-A-Fit**
## Bug/Feature description
Add a filter dialog to filter workout types based on user-selected choices. That way they can focus on viewing specific workout types

## Solution description
Integrated Material 3 Dialog, added in home screen events to dialog, logic is handled in home view model

## Code Validations 
[ ] PR is under 400 lines of code Insertion and Deletion  
[X] Files are Formatted and Optimized  
[X] Clean Architecture followed  
[X] No unnecessary comments or TODOs left in the codebase  
[ ] Unit Tests Added  
[X] Unit Tests NOT added and explanation why under exemption description  
[ ] UI Tests Added  
[X] UI Tests NOT added and explanation why under exemption description  
[ ] End to End Tests Added  
[X] End to End Tests NOT added and explanation why under exemption description  
[X] Screenshot of UI added  

## Exemption Definition
Explain if necessary the reason for any exemptions

## UI Screenshots 
filterDialogNew
fitJournalItemsSelected

## Test Case Screenshots
Place test Success creations here

### CODE REVIEW

File: app/src/main/java/com/example/fitjournal/components/dialogs/FilterJournalDialog.kt [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+                if (checkboxValue){
+                    if (!mutableWorkoutList.contains(workoutTypeEnum)) {
+                        mutableWorkoutList.add(workoutTypeEnum)
+                    }
+                }
+                if (!checkboxValue){
+                    if (mutableWorkoutList.contains(workoutTypeEnum)) {
+                        mutableWorkoutList.remove(workoutTypeEnum)
+                    }
+                }
```
Comment (SolidWolf) [inline app/src/main/java/com/example/fitjournal/components/dialogs/FilterJournalDialog.kt]: We can combine these if statements into:
```suggestion
                if (checkboxValue){
                    if (!mutableWorkoutList.contains(workoutTypeEnum)) {
                        mutableWorkoutList.add(workoutTypeEnum)
                    }
                } else {
                    if (mutableWorkoutList.contains(workoutTypeEnum)) {
                        mutableWorkoutList.remove(workoutTypeEnum)
                    }
                }
```

However, having nested if statements is considered bad practice so maybe we can do:
```suggestion
               when (checkboxValue) {
                   true -> {
                          if (!mutableWorkoutList.contains(workoutTypeEnum)) {
                                 mutableWorkoutList.add(workoutTypeEnum)
                          }
                    }
                   false -> {
                          if (mutableWorkoutList.contains(workoutTypeEnum)) {
                                 mutableWorkoutList.remove(workoutTypeEnum)
                          }
                   }
              }
```

File: app/src/main/java/com/example/fitjournal/components/dialogs/FilterJournalDialog.kt [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+                    shape = RoundedCornerShape(
+                        topStart = Spacing.spacing12,
+                        bottomStart = Spacing.spacing12
+                    )
+                )
+        ) {
+            Text(
+                text = stringResource(id = R.string.text_dismiss),
+                style = MaterialTheme.typography.bodyMedium.copy(fontWeight = FontWeight.Bold),
+                color = MaterialTheme.colorScheme.onSecondary
```
Comment (SolidWolf) [inline app/src/main/java/com/example/fitjournal/components/dialogs/FilterJournalDialog.kt:229]: use Coretheme
Kappa

File: app/src/main/java/com/example/fitjournal/screens/home/HomeScreenViewModel.kt [hunk 3]
```diff
@@ -70,15 +73,29 @@ class HomeScreenViewModel @Inject constructor() : ViewModel() {
...
+        val newFilteredList = currentFilterList.map {
+            FilterWorkoutModel(
+                isWorkoutSelected = filteredWorkoutList.contains(it.exerciseType),
+                exerciseType = it.exerciseType
+            )
+        }
+        updateHomeScreenState(newHomeScreenState = homeScreenState.copy(filterDialogList = newFilteredList))
+        //TODO("WE would want to now filter out the visible cards ")
+        Log.d("filter", "Filter works. List of cards should now show exercises based on filter enum types: $filteredWorkoutList")
+    }
```
Comment (SolidWolf) [inline app/src/main/java/com/example/fitjournal/screens/home/HomeScreenViewModel.kt:100]: don't forget to remove logs

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -1,3 +1,20 @@
...
+    <string name="title_weight_training">Weight Training</string>
+    <string name="title_calisthenics">Calisthenics</string>
+    <string name="title_cardio">Cardio</string>
+
+    // SubTitle Strings
+    <string name="subtitle_filter_workouts_description">Select which type of exercises to show in your Fit Journal</string>
+
+    // General Strings
+    <string name="text_dismiss">Dismiss</string>
+    <string name="text_confirm">Confirm</string>
```
Comment (SolidWolf) [inline app/src/main/res/values/strings.xml:18]: missing spanish translations 
( ͡° ͜ʖ ͡°)

**Review by SolidWolf — CHANGES_REQUESTED** [review]
address comment

File: app/src/main/java/com/example/fitjournal/components/dialogs/FilterJournalDialog.kt (hunk 1 above)
Comment (Alex-A-Fit) [reply app/src/main/java/com/example/fitjournal/components/dialogs/FilterJournalDialog.kt]: I dont think the nesting here creates any more complexity as the if statements themselves are very readable and self explanitory

File: app/src/main/java/com/example/fitjournal/components/dialogs/FilterJournalDialog.kt (hunk 2 above)
Comment (Alex-A-Fit) [inline app/src/main/java/com/example/fitjournal/components/dialogs/FilterJournalDialog.kt:228]: Canceled

File: app/src/main/java/com/example/fitjournal/screens/home/HomeScreenViewModel.kt (hunk 3 above)
Comment (Alex-A-Fit) [reply app/src/main/java/com/example/fitjournal/screens/home/HomeScreenViewModel.kt:100]: Keeping the log as the function itself is not finished

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (Alex-A-Fit) [inline app/src/main/res/values/strings.xml:18]: banned and canceled

**Review by SolidWolf — APPROVED** [review]

**Pull request merged**